lua/nvim-treesitter/locals.lua

```diff
@@ -89,4 +101,38 @@ function M.get_references(bufnr)
   return refs
 end
 
+
+--- Return all nodes in locals corresponding to a specific capture (like @scope, @reference)
+-- Works like M.get_references or M.get_scopes except you can choose the capture
+-- Can also be a nested capture like @definition.function to get all nodes defining a function
+function M.get_capture_matches(bufnr, capture_string, locals_kind)
+    if not string.sub(capture_string, 1,2) == '@' then
+      print('capture_string must start with "@"')
+      return
+    end
+
+    -- Build capture path: @definition.function -> ["definition", "function"]
```

Comment:
Could also be ["definition", "function", "node"] but not when we allow node groups for a capture.

Reply:
We already have this somewhere into `queries.lua`, it is called `split`, we should move it to `utils` then.

Reply:
I would rather replace the `split` in queries.lua by `vim.split`. For this I used the implementation of @steelsojka